apps/launcher/advancedpage.cpp

```diff
@@ -0,0 +1,89 @@
+#include "advancedpage.hpp"
+
+#include <components/files/configurationmanager.hpp>
+
+Launcher::AdvancedPage::AdvancedPage(Files::ConfigurationManager &cfg, Settings::Manager &engineSettings, QWidget *parent)
+        : QWidget(parent)
+        , mCfgMgr(cfg)
+        , mEngineSettings(engineSettings)
+{
+    setObjectName ("AdvancedPage");
+    setupUi(this);
+
+    loadSettings();
+}
+
+bool Launcher::AdvancedPage::loadSettings()
+{
+    // Game Settings
+    loadSettingBool(canLootDuringDeathAnimationCheckBox, "can loot during death animation", "Game");
+    loadSettingBool(followersAttackOnSightCheckBox, "followers attack on sight", "Game");
+    loadSettingBool(preventMerchantEquippingCheckBox, "prevent merchant equipping", "Game");
+    loadSettingBool(showEffectDurationCheckBox, "show effect duration", "Game");
+    loadSettingBool(showMeleeInfoCheckBox, "show enchant chance", "Game");
+    loadSettingBool(showMeleeInfoCheckBox, "show melee info", "Game");
+    loadSettingBool(showProjectileDamageCheckBox, "show projectile damage", "Game");
+
+    // Expected values are (0, 1, 2, 3)
+    int showOwnedIndex = mEngineSettings.getInt("show owned", "Game");
+    // Match the index with the option. Will default to 0 if invalid.
+    if (showOwnedIndex >= 0 && showOwnedIndex <= 3)
+        showOwnedComboBox->setCurrentIndex(showOwnedIndex);
+
+    // Input Settings
+    loadSettingBool(allowThirdPersonZoomCheckBox, "allow third person zoom", "Input");
+    loadSettingBool(grabCursorCheckBox, "grab cursor", "Input");
+    loadSettingBool(toggleSneakCheckBox, "toggle sneak", "Input");
+
+    // Other Settings
+    loadSettingBool(timePlayedCheckbox, "timeplayed", "Saves");
+
+    QString screenshotFormatString = QString::fromStdString(mEngineSettings.getString("screenshot format", "General")).toUpper();
+    if (screenshotFormatComboBox->findText(screenshotFormatString) == -1)
+        screenshotFormatComboBox->addItem(screenshotFormatString);
+    screenshotFormatComboBox->setCurrentText(screenshotFormatString);
```

Comment:
Looks like build fails with some versions of Qt (Qt4, I suppose):
`‘class QComboBox’ has no member named ‘setCurrentText’`

You can try to use the setItemText() instead.

Reply:
Do we really need to support Qt4? The last version [ended support in December 2015](https://blog.qt.io/blog/2015/05/26/qt-4-8-7-released/).

Reply:
I've heard this narrative many times and I'm tired of it.

Developer 1: Hey, this breaks X.
Developer 2: X is old, do we still need to support X?

If you break it, you fix it. We don't remove support for something because you were too lazy to update one line. If anything, we remove support after a poll, discussion, etc, but that's hardly related to this PR.

In this case, there is an outstanding issue with Qt5 + OSG not allowing threaded rendering. It's been discussed before in other places. And then there was [this](https://stackoverflow.com/questions/40863661/how-do-i-get-the-system-qfiledialog-in-qt5) - on Ubuntu, with Qt5 you can either get native file dialogs *or* menus, but not both :(

IMO, their end of support for Qt4 was very premature. Anyway, when the time comes, we should probably update our default choice to Qt5 *before* our support for Qt4 is removed, to give users time to adapt.

Reply:
@scrawl Sorry if that came across poorly. I was unaware that there had been prior discussion about it, and was merely wanting to raise the topic. My plan is still to fix this line when I get back to my home computer, and possibly raise a topic on the forums.